src/Implement/MainManager.cpp

```diff
@@ -0,0 +1,80 @@
+#include "MainManager.h"
+
+MainManager *MainManager::only_mainManager = nullptr;
+
+void SigHandle(int sig, siginfo_t *info, void *ctx) {
+  if (sig == SIGUSR1) {
+    mainManager.RemoveByValue(info->si_pid);
+  }
+}
+
+void MainManager::InitInstance(const uint16_t &port_num) {
+  this->port_num = port_num;
+  HttpListener listen_socket = HttpListener::GetInstance();
+  listen_socket.InitInstance(this->NewRequest, port_num);
+  listen_socket.StartListen();
+}
+
+MainManager *MainManager::GetInstance() {
+  if (only_mainManager == nullptr) {
+    only_mainManager = new MainManager();
+  }
+    return only_mainManager;
+}
+
+void MainManager::NewRequest(const int &socket_fd) {
+  pid_t child_pid = fork();
+  // means parent
+  if (child_pid > 0) {
+    socket_map[socket_fd] = child_pid;
+    struct sigaction act;
+    act.sa_sigaction = SigHandle;
+    sigemptyset(&act.sa_mask);
+    act.sa_flags = SA_SIGINFO;
+    sigaction(SIGUSR1, &act, NULL);
```

Comment:
Comment on why you need to catch SIGUSR1 or provide a reference to such comment.

Reply:
Done